src/analyzer/plugins/analyzerpluginsupport.cpp

```diff
@@ -0,0 +1,27 @@
+#include "analyzer/plugins/analyzerpluginsupport.h"
+
+QString AnalyzerPluginSupportInfo::matchAndSetPluginId(const QString& pluginIdToMatch) {
```

Comment:
set functions normaly set a member variable which is not the case here. 
How about renaming this to "matchOrGetDefaultPluginId" or just "matchPlugInID" or ...